lib/Makefile

```diff
@@ -0,0 +1,8 @@
+
+LIB=	ogit
+SHLIB_MAJOR=	1
```

Comment:
Major lib numbering on OpenBSD starts at 0. We also use a M.m numbering scheme.
Furthermore, in order for a shared library to be built on OpenBSD, there needs to be a file in the `lib` directory named `shlib_version` with the following contents:
```
major=0
minor=0
```
We would also need to find a way to keep that in sync.

Reply:
In other words, SHLIB_MAJOR=1 is a noop for us (but that's fine).

Reply:
What exactly handles `shlib_version`? I'm tempted to use SHLIB_* as the source of truth and generate shlib_version before the build if we can get away with it.

Reply:
https://github.com/openbsd/src/blob/2861400ecbfdff6825314e1c3461615957ee810d/share/mk/bsd.lib.mk#L11

Reply:
Noted in IRC; OpenBSD seems happy with directly defined SHLIB_MAJOR/SHLIB_MINOR as long as `shlib_version` isn't provided; said file will clobber previous major/minor.